### PR TITLE
Enhance TagNameUtils to handle the TagOverrideConfig

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
Also remove the following config classes (not really config):
- TagConfig
- OfflineTagConfig
- RealtimeTagConfig

NOTE: tenant should never have suffix; tag should always have suffix
Remove the support of mixed use of tenant/tag to prevent attaching wrong suffix

### CODE REVIEW

File: pinot-common/src/main/java/org/apache/pinot/common/config/TagNameUtils.java [hunk 1]
```diff
@@ -27,78 +28,93 @@ private TagNameUtils() {
...
   }
 
   public static boolean isRealtimeServerTag(String tagName) {
-    return tagName.endsWith(ServerType.REALTIME.toString());
+    return tagName.endsWith(REALTIME_SERVER_TAG_SUFFIX);
   }
 
-  public static boolean isBrokerTag(String tagName) {
-    return tagName.endsWith(TenantRole.BROKER.toString());
+  public static String getTenantFromTag(String tagName) {
```
Comment (jackjlli) [inline pinot-common/src/main/java/org/apache/pinot/common/config/TagNameUtils.java]: You can call `isOfflineServerTag`, `isRealtimeServerTag`, and `isBrokerTag` to reduce code redundancy.

File: pinot-common/src/main/java/org/apache/pinot/common/config/TagNameUtils.java [hunk 2]
```diff
@@ -27,78 +28,93 @@ private TagNameUtils() {
...
     }
   }
 
-  public static String getOfflineTagForTenant(String tenantName) {
-    if (tenantName == null) {
-      return TagNameUtils.getOfflineTagForTenant(DEFAULT_TENANT_NAME);
-    }
-    if (tenantName.endsWith(ServerType.OFFLINE.toString())) {
-      return tenantName;
+  public static String extractBrokerTag(TenantConfig tenantConfig) {
```
Comment (jackjlli) [inline pinot-common/src/main/java/org/apache/pinot/common/config/TagNameUtils.java]: Can `tenantConfig` be null?

Comment (Jackie-Jiang) [reply pinot-common/src/main/java/org/apache/pinot/common/config/TagNameUtils.java]: It won't be (everything not annotated with nullable are non-null)

File: pinot-common/src/main/java/org/apache/pinot/common/config/TagNameUtils.java (hunk 1 above)
Comment (Jackie-Jiang) [reply pinot-common/src/main/java/org/apache/pinot/common/config/TagNameUtils.java]: Simplified

**Pull request merged**